File: Carthage/Checkouts/PromiseKit/Extensions/QuartzCore/Tests/TestQuartzCore.m
@import PMKQuartzCore;
@import PromiseKit;
@import QuartzCore;
@import XCTest;

@implementation TestCALayer: XCTestCase

- (void)test {
    id ex = [self expectationWithDescription:@""];

    [[CALayer layer] promiseAnimation:[CAAnimation new] forKey:@"center"].then(^{
        [ex fulfill];
    });

    [self waitForExpectationsWithTimeout:1 handler:nil];
}

@end

File: Carthage/Checkouts/PromiseKit/Extensions/QuartzCore/Tests/TestQuartzCore.swift
import PMKQuartzCore
import PromiseKit
import QuartzCore
import XCTest

class TestCALayer: XCTestCase {
    func test() {
        let ex = expectation(description: "")

        CALayer().add(.promise, animation: CAAnimation()).done { _ in
            ex.fulfill()
        }

        wait(for: [ex], timeout: 10)
    }
}
